File: Test.Project.Library/Citizen.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Test.Project.Library
{
    public class Citizen
    {
        const int AGE_TO_BECOME_ADULT = 18;

        public Guid Id { get; set; }

        public string Name { get; set; }

        public string LastName { get; set; }
        
        private string _Document_Number;

        public string Document_Number {
            get { return _Document_Number; }
            set {
                float number;

                if (String.IsNullOrEmpty(value))
                    throw new Exception("Invalid blank document number");
                else if (value.Length > 11)
                    throw new Exception("Document number cannot have more than 11 characters");
                else if (value.Length < 11)
                    throw new Exception("Document number cannot have less than 11 characters");
                else if (!float.TryParse(value, out number))
                    throw new Exception("Invalid document number.");

                _Document_Number = value;
            }
        }

        public char _Sex;

        public char Sex { 
            get {return _Sex; } 
            set { 
                if(value != 'M' && value != 'F')
                    throw new Exception("Invalid Sex");

                _Sex = value;
            } 
        }

        public DateTime BirthDate { get; set; }

        public bool Save()
        {
            /*Emulate database call*/
            this.Id = Guid.NewGuid();
            return true;
        }

        public int Age()
        {
            var today = DateTime.Now;
            int age = today.Year - this.BirthDate.Year;
            return this.BirthDate > today.AddYears(-age) ? age : age--;
        }

        public bool IsAdult()
        {
            return this.Age() >= AGE_TO_BECOME_ADULT;
        }

        public enum SexType { 
            Male = 'M', 
            Female = 'F'
        }
    }
}

File: tests-js-example/citizen.js
var Citizen = function (citizen){
	this.id = citizen.id || '';
	this.name = citizen.name || '';
	this.lastName = citizen.lastName || '';
	this._documentNumber = citizen.documentNumber || '';
	this._sex = citizen.sex || '';
	this._birthDay = citizen.birthDay || '';

	this.sexType = {
		male : 'm',
		female : 'f'
	};
};

Object.defineProperty(Citizen.prototype, "sex", {
	  get: function() {return this._sex; },
	  set: function(value) { 

	  	if(!~['M', 'F'].indexOf(value.toUpperCase()))
	  		throw 'Invalid sex';
	  	else
	  		this._sex = value;
	  }
});

Object.defineProperty(Citizen.prototype, "documentNumber", {
	  get: function() {return this._documentNumber; },
	  set: function(value) { 

	  	if(!value)
	  		throw 'Invalid blank document number';
	  	else if (value.length > 11)
	  		throw 'Document number cannot have more than 11 characters';
	  	else if (value.length < 11)
	  		throw 'Document number cannot have less than 11 characters';
	  	else if (/[a-zA-Z]/.test(value))
	  		throw 'Invalid document number';
	  	else
	  		this._documentNumber = value;
	  }
});

//The best thing to do here is to use a library like momentjs to parse dates, but this would do 
Object.defineProperty(Citizen.prototype, "birthDay", {
	  get: function() {return this._birthDay; },
	  set: function(value) {
	  	if(value instanceof Date) 
	  		this._birthDay = value;
	  	else
	  		throw 'Invalid date';
	  		
	  }
});

//Async function
Citizen.prototype.save = function(callback){
	//*insert into the db here*
	this.id = newGuid();
	callback(true);
};

Citizen.prototype.age = function(){
    var ageDifMs = Date.now() - this.birthDay.getTime();
    var ageDate = new Date(ageDifMs); // miliseconds from epoch
    return Math.abs(ageDate.getUTCFullYear() - 1970);
};

Citizen.prototype.isAdult = function(){
    return this.age() > 18;
};

//http://stackoverflow.com/questions/105034/create-guid-uuid-in-javascript
var newGuid = function() {
  function s4() {
    return Math.floor((1 + Math.random()) * 0x10000)
      .toString(16)
      .substring(1);
  };
  return s4() + s4() + '-' + s4() + '-' + s4() + '-' +
    s4() + '-' + s4() + s4() + s4();
};

if(module)
	module.exports = Citizen;